### ISSUE TITLE
Distinguish between region and language

### ISSUE THREAD

**Issue by Miyayes** [brave-intl/publishers#3089]
## Description

Currently, we do not distinguish between language and region. Some of our features are determined by geographic region, such as available wallet services.

For instance, if a user's interface is in the Japanese language, then we treat them as being in the Japan region. However, a Japanese person might be a US resident—and therefore should have Uphold and Gemini options—but have their interface language set to Japanese.

## Solution

Distinguish between language and region: e.g., `ja_jp` and `en_us`, etc.

**Comment by Miyayes** [comment]
This includes strange edge cases where a Creator is Uphold-verified, but if they load into `?locale=ja`, they are in a mixed state that includes both PayPal (Japan only) and Uphold elements.

**Comment by Miyayes** [comment]
Add geo dropdown, warning messages when switching between geos, etc.

**Comment by OAuthBringer** [comment]
I suspect we would want to add something to the onboarding stage as well, part of onboarding auto-detects region/allows a user to set it/the language preferences.

**Comment by OAuthBringer** [comment]
WRT to the warning messages/switching piece we should determine what the business logic for switching from one location to another should be. I.e. from Japan to X or X to Japan.  Related to the edge case that was mentioned we'd probably want to default to disconnecting the wallet.

**Comment by tsmartt** [comment]
> disconnecting the wallet.

Yes, if you switch locales, we should warn that this will disconnect any non-Bitflyer wallets and vice-versa.

**Comment by OAuthBringer** [comment]
RE changing region and wallets:

[christopher]ents section on issue. Just show a confirmation modal / messaging to the user whenever they change geo, letting them know they'll be disconnected from custodian.  (edited)
